### PR TITLE
fix the update-nix script

### PR DESCRIPTION
**Pull request by manveru**
# Description

_A clear and concise description of what this pull request does or fixes._

# Proposed Solution

_**Optional** Explain how does this PR solves the problem stated in [Description](#Description). You can also enumerate different alternatives considered while approaching this task._

# Important Changes Introduced

_**Optional** Notice Reviewers about changes that were introduced while developing this task_

# Testing

_**Optional** Leave some recommendations should be useful while reviewers are testing this PR_

### CODE REVIEW

**Review by jonringer — APPROVED** [review]
LGTM

**Review by ten15bit — APPROVED** [review]

File: update-nix.sh [hunk 1]
```diff
@@ -11,30 +11,41 @@ usage() {
...
 fi
 
 echo "Determining new sha for sbt build, this can take several minutes to do a 'sbt compile'"
 
-NEW_SHA=$(nix-build -E 'with import ./. {}; deps.overrideAttrs( _: { outputHash = "0000000000000000000000000000000000000000000000000000"; })' 2>&1 | grep "  got: " | head -n 1 | sed -r 's/\s+got:\s+//' | xargs nix-hash --to-base32 --type sha256 )
+output="$(nix build --impure --expr '(builtins.getFlake (toString ./.)).legacyPackages.x86_64-linux.mantis-hash { ref = "fix-update-nix"; rev = "7e86800be34c328c3149a472e8a79e90534c8838"; }' 2>&1 || true)"
+
+NEW_SHA=$(
+ echo "$output" \
+   | awk '/^     got: / { print $2 }' foo
```
Comment (jonringer) [inline update-nix.sh]:
```suggestion
   | awk '/^     got: / { print $2 }'
```
```
awk: fatal: cannot open file `foo' for reading (No such file or directory)
```

File: update-nix.sh [hunk 2]
```diff
@@ -11,30 +11,41 @@ usage() {
...
-if [ "$1" == "-h" -o "$1" == "--help" ]; then
+if [ "$1" == "-h" ] || [ "$1" == "--help" ]; then
         usage
         exit 1
 fi
 
 echo "Determining new sha for sbt build, this can take several minutes to do a 'sbt compile'"
 
-NEW_SHA=$(nix-build -E 'with import ./. {}; deps.overrideAttrs( _: { outputHash = "0000000000000000000000000000000000000000000000000000"; })' 2>&1 | grep "  got: " | head -n 1 | sed -r 's/\s+got:\s+//' | xargs nix-hash --to-base32 --type sha256 )
+output="$(nix build --impure --expr '(builtins.getFlake (toString ./.)).legacyPackages.x86_64-linux.mantis-hash { ref = "fix-update-nix"; rev = "7e86800be34c328c3149a472e8a79e90534c8838"; }' 2>&1 || true)"
```
Comment (manveru) [inline update-nix.sh]: ref and rev should be take from the git status, not hardcoded.

**Review by infinisil — APPROVED** [review]
LGTM!

**Review by ten15bit — APPROVED** [review]

**Pull request merged**